ingestion/src/metadata/ingestion/source/pipeline/dagster/client.py

```diff
@@ -0,0 +1,137 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Wrapper module of DagsterGraphQLClient client
+"""
+
+import traceback
+from typing import List, Optional
+
+from dagster_graphql import DagsterGraphQLClient
+from gql.transport.requests import RequestsHTTPTransport
+
+from metadata.generated.schema.entity.services.connections.pipeline.dagsterConnection import (
+    DagsterConnection,
+)
+from metadata.ingestion.source.pipeline.dagster.models import (
+    DagsterPipeline,
+    GraphOrError,
+    GraphOrErrorModel,
+    Node,
+    PipelineOrErrorModel,
+    RepositoriesOrErrorModel,
+)
+from metadata.ingestion.source.pipeline.dagster.queries import (
+    DAGSTER_PIPELINE_DETAILS_GRAPHQL,
+    GRAPHQL_QUERY_FOR_JOBS,
+    GRAPHQL_RUNS_QUERY,
+)
+from metadata.utils.helpers import clean_uri
+from metadata.utils.logger import ometa_logger
+
+logger = ometa_logger()
+
+
+class DagsterClient:
+    """
+    Wrapper to DagsterGraphQLClient
+    """
+
+    def __init__(self, config: DagsterConnection):
+        url = clean_uri(config.host)
+        self.client = DagsterGraphQLClient(
+            url,
+            transport=RequestsHTTPTransport(
+                url=f"{url}/graphql",
+                headers={"Dagster-Cloud-Api-Token": config.token.get_secret_value()}
+                if config.token
+                else None,
+                timeout=config.timeout,
+            ),
+        )
+
+    def get_run_list(self) -> Optional[List[Node]]:
+        """
+        List all the pipeline runs
+        """
+        try:
+            result = self.client._execute(  # pylint: disable=protected-access
+                DAGSTER_PIPELINE_DETAILS_GRAPHQL
+            )
+            result = RepositoriesOrErrorModel(**result)  # pylint: disable=not-a-mapping
+            return result.repositoriesOrError.nodes
+        except ConnectionError as conerr:
+            logger.error(f"Cannot connect to dagster client {conerr}")
+            logger.debug(f"Failed due to : {traceback.format_exc()}")
```

Comment:
I believe our usual approach is:
1. First logger.debug
2. Then logger.error

Also the debug has some extra spaces `Failed due to:...` instead of `Failed due to : ...`